src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,125 @@
+<!--
+***********************************************************************************************
+Microsoft.Android.Sdk.Bindings.Gradle.targets
+
+This file contains MSBuild targets that support building and operating on Android Gradle projects.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.Gradle"              AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <_AndroidGradleDefaultOutputPathRoot>$(MSBuildProjectDirectory)/$(IntermediateOutputPath)gradle/</_AndroidGradleDefaultOutputPathRoot>
+    <_AndroidGradleBuildDirInitScriptPath>$(_AndroidGradleDefaultOutputPathRoot)net.android.init.gradle.kts</_AndroidGradleBuildDirInitScriptPath>
+    <_BuildAndroidGradleProjectsStamp>$(_AndroidStampDirectory)_BuildAndroidGradleProjects.stamp</_BuildAndroidGradleProjectsStamp>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <AndroidGradleProject>
+      <Bind>true</Bind>
+      <Configuration>Release</Configuration>
+      <ModuleName></ModuleName>
+      <OutputPath></OutputPath>
+      <Pack>true</Pack>
+      <ReferenceLibraryOutputs>true</ReferenceLibraryOutputs>
+      <Visible>false</Visible>
+    </AndroidGradleProject>
+  </ItemDefinitionGroup>
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs"
+        Condition=" '@(AndroidGradleProject->Count())' != '0' ">
+    <ItemGroup>
+      <_AllGradleFiles Include="$([System.IO.Directory]::GetFiles('%(AndroidGradleProject.RootDir)%(AndroidGradleProject.Directory)', '*.*', System.IO.SearchOption.AllDirectories))"
+          Condition= " Exists('%(AndroidGradleProject.RootDir)%(AndroidGradleProject.Directory)') "/>
+      <_GradleInputs Include="@(_AllGradleFiles)"
+          Condition="  '%(Extension)' == '.java'
+                    or '%(Extension)' == '.kt'
+                    or '%(Extension)' == '.kts'
+                    or '%(Extension)' == '.gradle'
+                    or '%(Extension)' == '.properties'
+                    or '%(Extension)' == '.xml' " />
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="@(_AndroidMSBuildAllProjects)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_CalculateAndroidGradleProjectOutputPath"
+        Condition=" '@(AndroidGradleProject->Count())' != '0' "
+        Outputs="%(AndroidGradleProject.Identity)" >
+    <Hash
+        ItemsToHash="%(AndroidGradleProject.Identity)"
+        IgnoreCase="false">
```

Comment:
Given that `%(Identity)` is a filesystem path, `IgnoreCase` should *probably* be True, as that'll be the filesystem semantics on Windows and macOS.